### PR TITLE
feat: add SecureHeaders filter

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add filter that add common security related filters to response object

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Filters/SecureHeaders.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        // https://owasp.org/www-project-secure-headers/#x-permitted-cross-domain-policies
+        'X-Permitted-Cross-Domain-Policies' => 'none',
+
+        // https://owasp.org/www-project-secure-headers/#referrer-policy
+        'Referrer-Policy' => 'same-origin',
+
+        // https://owasp.org/www-project-secure-headers/#x-xss-protection
+        // If you do not need to support legacy browsers, it is recommended that you use
+        // Content-Security-Policy without allowing unsafe-inline scripts instead.
+        // 'X-XSS-Protection' => '1; mode=block',
```
Comment (paulbalandan) [inline system/Filters/SecureHeaders.php:44]: If I want to activate this without editing this system file, how will I do it?

Comment (kenjis) [reply system/Filters/SecureHeaders.php:44]: Extend and override it.

Comment (paulbalandan) [reply system/Filters/SecureHeaders.php:44]: Hmm ok. Maybe we should add these options in the docs and tell which are activated by default.

Comment (kenjis) [reply system/Filters/SecureHeaders.php:44]: Added the explanation in the docs.

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**